pkg/rpc/type.go

```diff
@@ -0,0 +1,286 @@
+}
+
+// Create a new `FetchCluster` response
+func NewFetchClusterResponse(res *curppb.FetchClusterResponse) *fetchClusterResponse {
+	return &fetchClusterResponse{
+		FetchClusterResponse: res,
+	}
+}
+
+// Get all members addresses
+func (r *fetchClusterResponse) IntoMembersAddrs() map[serverId][]string {
+	members := map[serverId][]string{}
+	for _, member := range r.Members {
+		members[member.Id] = member.Addrs
+	}
+	return members
+}
+
+// Propose request
+type proposeRequest struct {
+	*curppb.ProposeRequest
+}
+
+// Create a new `Propose` request
+func NewProposeRequest(
+	proposeId *curppb.ProposeId,
+	cmd *xlinepb.Command,
+	clusterVersion uint64,
+) *proposeRequest {
+	bcmd, err := proto.Marshal(cmd)
+	if err != nil {
+		panic(err)
+	}
+	return &proposeRequest{
+		ProposeRequest: &curppb.ProposeRequest{
+			ProposeId:      proposeId,
+			Command:        bcmd,
+			ClusterVersion: clusterVersion,
+		},
+	}
+}
+
+// Propose response
+type proposeResponse struct {
+	*curppb.ProposeResponse
+}
+
+// Create a new `Propose` response
+func NewProposeResponse(res *curppb.ProposeResponse) *proposeResponse {
+	return &proposeResponse{
+		ProposeResponse: res,
+	}
+}
+
+// Deserialize result in response
+func (r *proposeResponse) DeserializeResult() (*ProposeResult, *CurpError) {
+	exeResult := xlinepb.CommandResponse{}
+	exeErr := xlinepb.ExecuteError{}
+
+	if r.Result == nil {
+		return nil, nil
+	}
+	switch cmdResult := r.Result.Result.(type) {
+	case *curppb.CmdResult_Ok:
+		err := proto.Unmarshal(cmdResult.Ok, &exeResult)
+		if err != nil {
+			return nil, NewCurpError(err)
+		}
+		return &ProposeResult{ExeResult: &exeResult}, nil
+	case *curppb.CmdResult_Error:
+		err := proto.Unmarshal(cmdResult.Error, &exeErr)
+		if err != nil {
+			return nil, NewCurpError(err)
+		}
+		return &ProposeResult{ExeErr: &exeErr}, nil
+	default:
+		panic("unknown result type")
+	}
+}
+
+// Wait synced request
+type waitSyncedRequest struct {
+	*curppb.WaitSyncedRequest
+}
+
+// Create a new `WaitSynced` request
+func NewWaitSyncedRequest(id *curppb.ProposeId, clusterVersion uint64) *waitSyncedRequest {
+	return &waitSyncedRequest{
+		WaitSyncedRequest: &curppb.WaitSyncedRequest{
+			ProposeId:      id,
+			ClusterVersion: clusterVersion,
+		},
+	}
+}
+
+// Create a new `WaitSynced` response
+type WaitSyncedResponse struct {
+	*curppb.WaitSyncedResponse
+}
+
+// Create a new `WaitSynced` response
+func NewWaitSyncedResponse(res *curppb.WaitSyncedResponse) *WaitSyncedResponse {
+	return &WaitSyncedResponse{
+		WaitSyncedResponse: res,
+	}
+}
+
+// Deserialize result in response
+func (r *WaitSyncedResponse) DeserializeResult() (*WaitSyncedResult, *CurpError) {
+	// according to the above methods, we can only get the following response union
+	// ER: Some(OK), ASR: Some(OK)  <-  WaitSyncedResponse::new_success
+	// ER: Some(Err), ASR: None     <-  WaitSyncedResponse::new_er_error
+	// ER: Some(OK), ASR: Some(Err) <- WaitSyncedResponse::new_asr_error
```

Comment:
Please update these comments, `Some` and `None` are Rust keywords

Reply:
I've updated.